cni/pkg/plugin/plugin_test.go

```diff
@@ -507,6 +509,46 @@ func TestCmdAddExcludePodWithIstioInitContainer(t *testing.T) {
 	}
 }
 
+func TestCmdAddWithProxySidecarAmbientEnabledWithIstioInitContainer(t *testing.T) {
```

Comment:
The test code is fine (ty for having a go at it), but this added test is redundant with the existing test `TestCmdAddExcludePodWithIstioInitContainer`.

The presence of an init container is not checked or used to exclude a pod, the annotation is.

(this is a little weird, but you can't get the one without the other in sidecar mode so it should be fine).

Reply:
Thank you for reviewing it. What I planned, was to check if a sidecar injected in a pod is added to the mesh or not.
pod.ObjectMeta.Annotations[label.SidecarInject.Name] = "true"
pod.ObjectMeta.Annotations[label.Sidecar.Status] = "true"
Only with these annotations without configuring namespace, shouldn't the pod be added to mesh?(as ambient is not enabled)

Reply:
Also I believe there's no need to add the code of this branch to master, I better work on the helm_upgrade_test.go instead of adding tests to plugin_test.go in cni.

Reply:
> pod.ObjectMeta.Annotations[label.SidecarInject.Name] = "true"
> pod.ObjectMeta.Annotations[label.Sidecar.Status] = "true"
> Only with these annotations without configuring namespace, shouldn't the pod be added to mesh?(as ambient is not enabled)

If ambient is not enabled, no pods should be added to the ambient mesh.

If ambient is enabled, no pods with injected sidecar proxies or sidecar labels should be added to the ambient mesh.